puremagic/main.py

```diff
@@ -159,33 +159,51 @@ def _identify_all(header: bytes, footer: bytes, ext=None) -> List[PureMagicWithC
     for matched in matches:
         if matched.byte_match in multi_part_dict:
             for magic_row in multi_part_dict[matched.byte_match]:
-                start = magic_row.offset
-                end = magic_row.offset + len(magic_row.byte_match)
-                if magic_row.offset < 0:
-                    match_area = footer[start:end] if end != 0 else footer[start:]
-                    if match_area == magic_row.byte_match:
+                if "###REGEX###" in magic_row.name:
+                    import re
+
+                    if not magic_row.offset == 0:
+                        scan_bytes = header[0 : magic_row.offset]
+                    else:
+                        scan_bytes = header
```

Comment:
```suggestion
                     scan_bytes = header[0 : magic_row.offset] if magic_row.offset else header
```

Reply:
Thanks for that, my Python skills are like me, ugly but functional. 🙂

This PR will likely close for reasons discussed above, but the chances are some of it will come back in one form or another in V2.0